lib/gitgitgadget.ts

```diff
@@ -260,14 +260,19 @@ export class GitGitGadget {
         // update work repo from base
         await this.updateNotesAndPullRef(pr.number, previousTag);
 
-        // Remove template from description
-        let prTemplate =
-            await git(["show",
-                       "upstream/master:.github/PULL_REQUEST_TEMPLATE.md"],
-                      { workDir: this.workDir });
-        // github uses \r\n so make sure it is set
-        prTemplate = prTemplate.replace(/([^\r])\n/, "$1\r\n");
-        const prBody = pr.body.replace(prTemplate, "");
+        // Remove template from description (if template exists)
+        let prBody: string;
+        try {
+            let prTemplate =
+                await git(["show",
+                           "upstream/master:.github/PULL_REQUEST_TEMPLATE.md"],
```

Comment:
It occurred to me last night that PR templates might differ depending on the base branch. Do you know?

Reply:
Not sure what you asking.  This change assumes the base branch is master.  If it was not, then we would need to parse the `baseLabel` to get the name.  If I cloned and changed the base branch in my repo and did a pull request to my repo I could have a different template.

Reply:
> This change assumes the base branch is master.

Many are not, though...

> If it was not, then we would need to parse the `baseLabel` to get the name.

Yes, I think we will want to do that. IIRC the `baseLabel` starts with the `${repositoryOwner}:` prefix, though, so we will want to be careful to use the correct thing there. Maybe `baseCommit` would be a better idea?

> If I cloned and changed the base branch in my repo and did a pull request to my repo I could have a different template.

True. But how common is it that somebody opens a PR, _forgets_ to remove the template text, _and_ is savvy enough to change the target branch?

Reply:
> > This change assumes the base branch is master.
> 
> Many are not, though...

We may be talking about different bases here.  I was referring to the base/default branch for the repo, which in the case of git is master.  That is my understanding of what GitHub uses for the template.  Are you referring to the base for the pull request, which I agree may be different?

> > If it was not, then we would need to parse the `baseLabel` to get the name.
> 
> Yes, I think we will want to do that. IIRC the `baseLabel` starts with the `${repositoryOwner}:` prefix, though, so we will want to be careful to use the correct thing there. Maybe `baseCommit` would be a better idea?

It might actually be `base.repo.default_branch` that is needed.  It can be added to `IPullRequestInfo`.  Can this be deferred until it is actually needed?  Otherwise, the fetch of upstream will also need to ensure it is pulling that branch.

Reply:
> > > This change assumes the base branch is master.
> > 
> > 
> > Many are not, though...
> 
> We may be talking about different bases here. I was referring to the base/default branch for the repo, which in the case of git is master. That is my understanding of what GitHub uses for the template.

I am not sure that the `master` version is used always, even when the user targets a different branch. Is this documented somewhere?

If it is _always_ the `master` version, then I retract my suggestion and declare the current version fine ;-)

Reply:
>  Is this documented somewhere?

It is ;-) https://help.github.com/en/github/building-a-strong-community/about-issue-and-pull-request-templates#pull-request-templates says:

> You must create templates on the repository's default branch. Templates created in other branches are not available for collaborators to use.

So: I retract my suggestion to look at `baseCommit` ;-) And since we are mirroring git/git's `master` into gitgitgadget/git always, `upstream/master` is good enough and you do not have to do anything differently.

Sorry for the noise!

Reply:
I recently found out if you push a new repo to github with branch foo having commits and no commits on master that branch foo is the default branch on github.  Had to reset in the settings.  😮

Reply:
Oh yes, that happened to me, too. I was quite surprised by that, but it kinda makes sense.